### PR TITLE
[CI] Onboard pytest and deprecate nose for Automation Test

### PR DESCRIPTION
**Pull request by haroldrandom**
**Description**  
1. Onboard new azdev to run automation test.
2. Onboard pytest to run Automation Test
3. Deprecate nose test framework

**Testing Guide**  
1. test incremental test on network package https://github.com/Azure/azure-cli/pull/13173,
  1.1 it ran 132 tests under profile latest
  1.2 it ran 71 tests under profile 2018
2. test incremental test on compute package https://github.com/Azure/azure-cli/pull/13175, it ran all tests under latest profile for vm.
3. test incremental test on storage package https://github.com/Azure/azure-cli/pull/13176, it ran all tests under latest profile for storage.
4. test incremental test on keyvault package https://github.com/Azure/azure-cli/pull/13177, it ran all tests under latest profile for keyvault 
5. test incremental test on resource https://github.com/Azure/azure-cli/pull/13178, it ran all tests under latest profile for resource
6. test core https://github.com/Azure/azure-cli/pull/13179, 
  6.1 it rall all 1936 tests under profile latest
  6.2 it ran all 610 tests under profile 2019-03-01-hybrid
  6.3 it ran all 635 tests under profile 2018-03-01-hybrid
  6.4 it ran all 314 tests under profile 2017-03-09-profile
7. test network https://github.com/Azure/azure-cli/pull/13181 for profile 2018-03-01-hybrid, it ran all tests including core as design.
8. test vm https://github.com/Azure/azure-cli/pull/13182 for profile 2019-03-01-hybrid, it ran all tests including core as desgin.
9. test network and vm https://github.com/Azure/azure-cli/pull/13185 for profile latest, it ran all vm and network tests.

**History Notes**  

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

File: azure-pipelines.yml [hunk 1]
```diff
@@ -407,142 +407,131 @@ jobs:
     - bash: ./scripts/ci/test_profile_integration.sh
       displayName: 'Run Integration Test against Profiles'
 
-- job: RunAutomationPython3_6
-  displayName: Run Automation Python 3.6
+- job: AutomationTest
+  displayName: Automation Test (Profile Latest)
   dependsOn: BuildPythonWheel
-  timeoutInMinutes: 90
+  timeoutInMinutes: 120
```
Comment (MyronFanQiu) [inline azure-pipelines.yml:413]: should it be incremental?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -407,142 +407,131 @@ jobs:
...
+        azdev --version
+
+        azdev setup -c ./
+      displayName: Setup Env
+    - bash: |
+        set -ev
+
+        source env/bin/activate
+
+        azdev test --series --profile 2019-03-01-hybrid
```
Comment (MyronFanQiu) [inline azure-pipelines.yml:493]: curious about this part. Why do we not check 
```if [[ "$(System.PullRequest.TargetBranch)" != "" ]]; then``` for other profile?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -407,142 +407,131 @@ jobs:
...
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+        python -m venv env
+        chmod +x env/bin/activate
+        source env/bin/activate
 
-- job: RunAutomation20180301
-  displayName: Run Automation, Profile 2018-03-01
-  dependsOn: BuildPythonWheel
+        pip install -q azdev
+        azdev --version
```
Comment (qianwens) [inline azure-pipelines.yml:438]: duplicate scripts in different profile, please wrap them in a script file or common variable in yml?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -566,33 +555,6 @@ jobs:
...
-  - bash: |
-      pip install --force-reinstall wheel
-  - task: Bash@3
-    displayName: 'Run Test Automation'
-    inputs:
-      targetType: 'filePath'
-      filePath: ./scripts/ci/test_automation.sh
-    env:
-      AZURE_CLI_TEST_TARGET_PROFILE: '2018-03-01'
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
```
Comment (qianwens) [inline azure-pipelines.yml:594]: why cannot we reuse these two environment?

File: azure-pipelines.yml [hunk 5]
```diff
@@ -407,142 +407,131 @@ jobs:
...
+
+        azdev setup -c ./
+      displayName: Setup Env
+    - bash: |
+        set -ev
+
+        source env/bin/activate
+
+        azdev test --series --profile 2018-03-01-hybrid
+      displayName: "Automation Test"
```
Comment (qianwens) [inline azure-pipelines.yml:534]: for 2018-03-01-hybrid?

File: azure-pipelines.yml (hunk 1 above)
Comment (haroldrandom) [reply azure-pipelines.yml:413]: pytest take more time to run all tests (1936), so, have to increase this timeout limit. And Will be fixed in the future to optimize the time cost.

File: azure-pipelines.yml (hunk 5 above)
Comment (haroldrandom) [reply azure-pipelines.yml:534]: You mean suggesting change the `displayName` value to `Automation Test for 2018-03-01-hybrid` ?

File: azure-pipelines.yml (hunk 2 above)
Comment (haroldrandom) [reply azure-pipelines.yml:493]: We could, but the number of tests for other profiles are small, so I decide to run them all as we discussed.

File: azure-pipelines.yml (hunk 1 above)
Comment (fengzhou-msft) [reply azure-pipelines.yml:413]: What's the time cost now?

File: azure-pipelines.yml [hunk 6]
```diff
@@ -407,142 +407,131 @@ jobs:
...
-    env:
-      ADO_PULL_REQUEST_TARGET_BRANCH: $(System.PullRequest.TargetBranch)
+    - task: UsePythonVersion@0
+      inputs:
+        versionSpec: '$(python.version)'
+      displayName: 'Use Python $(python.version)'
+    - bash: |
+        set -ev
+
+        git fetch origin dev
```
Comment (fengzhou-msft) [inline azure-pipelines.yml:477]: why use a git fetch here?

Comment (haroldrandom) [reply azure-pipelines.yml:477]: Removed since git checkout in ADO will fetch all target branches.

File: azure-pipelines.yml (hunk 1 above)
Comment (haroldrandom) [reply azure-pipelines.yml:413]: 1H 49m from this [build ](https://dev.azure.com/azure-sdk/public/_build/results?buildId=370756&view=logs&j=74095127-2a27-5370-37ed-15a4193f243f&t=74095127-2a27-5370-37ed-15a4193f243f) from this PR https://github.com/Azure/azure-cli/pull/13179

File: azure-pipelines.yml (hunk 2 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: Now, incremental test apply on all profiles since I use template.

File: azure-pipelines.yml (hunk 3 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: I searched and found the ADO template only works for steps not for job. So, I moved those similar steps into a single file `.azure-pipelines/template/automation_test.yml`.

File: azure-pipelines.yml (hunk 4 above)
Comment (haroldrandom) [reply azure-pipelines.yml:594]: They have became parameters while instantiating the template `.azure-pipelines/template/automation_test.yml`.

File: azure-pipelines.yml (hunk 5 above)
Comment (haroldrandom) [reply azure-pipelines.yml]: Changed.

**Comment by haroldrandom** [comment]
buggy ADO...the actual output on ADO portal shows all jobs are passed in Attemp 2 but on Github it shows the result of Attemp 1
image

**Pull request merged**